### ISSUE TITLE
Collapse: Make id prop optional

### ISSUE THREAD

**Issue by MarcusWichelmann** [bootstrap-vue/bootstrap-vue#1648]
The collapse component allows me to link it's opening-state to a button using `v-model` which is especially useful when using `BCollapse` in a dynamic list of custom components.
But even when this works completly fine, Vue.js reports `[Vue warn]: Missing required prop: "id"` because of the "id"-prop being marked as "required":
https://github.com/bootstrap-vue/bootstrap-vue/blob/dev/src/components/collapse/collapse.js#L58

I cannot just set the id propert to `something-INDEX` because I don't have any index or id inside of this component wich I could use for that.

**Comment by Diaskhan** [comment]
@MarcusWichelmann 
Id is required beacuse code must know id of component to hide and show content!
**Architecure by design**

**Comment by Diaskhan** [comment]
Friends issue could  close for now! There no solution for this !

**Comment by baram204** [comment]
Hey! @Diaskhan  this warn even showed up when I use `<b-list-group>`

I gave id property & it's value, but It keeps prints warn!!

1. code already know id 
2. hide and show elements word well.
3. It means they don't need any additional information.

so, why `b-collapse(b-list-group)` required me the data already they have?

and why they don't satisfy data(id prop and value) I gave them??

the code links [here](https://github.com/baram204/electon-vue-fitbit-toy-project/blob/master/src/renderer/components/sidebar/TreeMenu.vue)
